### PR TITLE
Dronekit improvements

### PR DESCRIPTION
**Pull request by sanderux**
Simplifying mode change and apply new json mission format

### CODE REVIEW

File: integrationtests/python_src/px4_it/dronekit/MissionCheck.py [hunk 1]
```diff
@@ -227,11 +162,8 @@ def listener(self, name, home_position):
 missionlist = upload_mission(import_mission_filename)
 time.sleep(2)
 
-# set mission mode the hard way
-vehicle._master.mav.command_long_send(vehicle._master.target_system, vehicle._master.target_component,
-                                           mavutil.mavlink.MAV_CMD_DO_SET_MODE, 0,
-                                           MAV_MODE_AUTO,
-                                           0, 0, 0, 0, 0, 0)
+# set mission mode
+vehicle.mode = VehicleMode("MISSION")
```
Comment (LorenzMeier) [inline integrationtests/python_src/px4_it/dronekit/MissionCheck.py]: This won't work on the current pyMAVLink version - can we keep the "hard way" until that's fully merged?

Comment (sanderux) [inline integrationtests/python_src/px4_it/dronekit/MissionCheck.py]: Works fine here with latest dronekit

Comment (LorenzMeier) [inline integrationtests/python_src/px4_it/dronekit/MissionCheck.py]: ok.

Comment (sanderux) [inline integrationtests/python_src/px4_it/dronekit/MissionCheck.py]: with pymavlink 2.0.9

**Comment by LorenzMeier** [comment]
Can you enable the CI test in the test script so that its actually being run? Or did I miss that it's already running?

**Comment by sanderux** [comment]
@dagar i don't have the SITL docker image running so i cannot test this. could you add this?
@LorenzMeier maybe merge first?

**Comment by LorenzMeier** [comment]
We can merge first. I'll send a new PR which just enables it in CI.

**Comment by sanderux** [comment]
It does rely on https://github.com/ArduPilot/pymavlink/pull/42

**Pull request merged**